src/widgets/radio.jsx

```diff
@@ -49,88 +194,62 @@ var BaseRadio = React.createClass({
                     <$_>Select all that apply.</$_>
                 </div>}
             {this.props.choices.map(function(choice, i) {
+                var classSet = { "inline": !this.props.onePerLine };
 
-                var content = <div>
-                        {choice.content}
-                    </div>;
+                // True if we're in review mode and a clue is available
+                var reviewModeClues = rubric && rubric.choices[i].clue;
+                var showClue = choice.checked &&
+                    (exerciseClues || reviewModeClues);
+
+                var element = Choice;
+                var elementProps = {
+                    ref: `radio${i}`,
+                    checked: choice.checked,
+                    clue: choice.clue,
+                    content: choice.content,
+                    disabled: this.props.apiOptions.readOnly,
+                    groupName: radioGroupName,
+                    showClue: showClue,
+                    type: inputType,
+                    onChecked: (checked) => {
+                        this.checkOption(i, checked);
+                    }
+                }
+
+                if (choice.isNoneOfTheAbove) {
+                    element = ChoiceNoneAbove;
+                    _.extend(elementProps, { showContent: choice.correct });
```

Comment:
I'm fine with this (and we've done this in other places, I think). I think focusing on readability over efficiency (at least for things that are reasonably fast already) is the right call :).

I think there are pros and cons for which is more readable--we've done this elsewhere, and it's somewhat common for changing multiple properties, which is good, but we also use a lot of `_.extend({},` for immutability, which this could be mistaken for. I'm happy with whichever you find more readable.